types/externs.d.ts

```diff
@@ -94,6 +94,17 @@ declare global {
     [K in keyof T as KebabToCamelCase<K>]: T[K];
   }
 
+  type TagNameToElement<
+    Tag extends string
+  > = Tag extends keyof HTMLElementTagNameMap
+    ? HTMLElementTagNameMap[Tag]
+    : Tag extends keyof SVGElementTagNameMap
+    ? SVGElementTagNameMap[Tag]
+    : HTMLElement; // typed-query-selector use Element here, which we don't want.
```

Comment:
I think we should maybe do the right thing and have it be `Element` here. Maybe we stuck to a default `HTMLElement` because too many things needed to be fixed simultaneously?

Playing with it for a few minutes to see how bad the transition would be and I already found a bug in our renderer :)

https://github.com/GoogleChrome/lighthouse/blob/b2aa4804578f5ed5a3b52d2fe13b1bd04e7aacd0/lighthouse-core/report/html/renderer/category-renderer.js#L146-L153

is allowed to return `HTMLElement` because `dom.find()` is defaulting to returning `HTMLElement`, but `.lh-chevron` is an `SVGElement`:

https://github.com/GoogleChrome/lighthouse/blob/b2aa4804578f5ed5a3b52d2fe13b1bd04e7aacd0/lighthouse-core/report/html/templates.html#L35-L37

As a result, when we try to add a `title` to the chevron for a `"Show audits"` tooltip:

https://github.com/GoogleChrome/lighthouse/blob/b2aa4804578f5ed5a3b52d2fe13b1bd04e7aacd0/lighthouse-core/report/html/renderer/category-renderer.js#L291-L292

the property is set but it doesn't do anything...because `SVGElement` doesn't support `title` for tooltips (I guess you need [`<title>`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title) for that or just set it on a parent `HTMLElement`, like we do for the [chevron on perf audits](https://github.com/GoogleChrome/lighthouse/blob/b2aa4804578f5ed5a3b52d2fe13b1bd04e7aacd0/lighthouse-core/report/html/templates.html#L137))

Reply:
good catch, changed to use Element as fallback.

required adding a lot of `div` prefixes